### PR TITLE
docs: Add Task Count table per language

### PR DESCRIPTION
**Pull request by isaac-chung**
Addresses https://github.com/embeddings-benchmark/mteb/discussions/686

- Add task count per language as a separate table in `tasks.md`
- Languages currently do not have associated families or genus. That can be added in a separate PR if needed. 

```
python docs/create_tasks_table.py
```

### CODE REVIEW

File: docs/create_tasks_table.py [hunk 1]
```diff
@@ -56,15 +65,49 @@ def create_tasks_table(tasks: list[mteb.AbsTask]) -> str:
     return table
 
 
-def insert_table(file_path, table):
-    """Insert table in the in <!-- TABLE START --> and <!-- TABLE END -->"""
+def create_task_lang_table(tasks: list[mteb.AbsTask]) -> str:
+    
+    table_dict = {}
+    ## Group by language. If it is a multilingual dataset, 1 is added to all languages present.
+    ## For now, we skip cross lingual datasets.
```
Comment (isaac-chung) [inline docs/create_tasks_table.py:72]: Was discussing this with @imenelydiaker: A cross-lingual dataset would have 2 languages, but it's still 1 dataset. However, we count each language within a multilingual dataset. Wanted some feedback on this.

File: docs/create_tasks_table.py [hunk 2]
```diff
@@ -2,8 +2,16 @@
 
 import re
```
Comment (mrshu) [inline docs/create_tasks_table.py:3]: @isaac-chung just FYI: this doesn't seem to be used anywhere in the code

File: docs/create_tasks_table.py [hunk 3]
```diff
@@ -2,8 +2,16 @@
...
 from pathlib import Path
+from typing import get_args
 
 import mteb
+from mteb.abstasks.TaskMetadata import PROGRAMMING_LANGS, TASK_TYPE
+
+
+def is_task_to_skip(task: mteb.AbsTask):
```
Comment (mrshu) [inline docs/create_tasks_table.py:11]:
```suggestion
def is_task_to_skip(task: mteb.AbsTask):
        """Determine whether a task should be omitted from reporting."""
```

File: docs/create_tasks_table.py [hunk 4]
```diff
@@ -56,15 +64,48 @@ def create_tasks_table(tasks: list[mteb.AbsTask]) -> str:
...
+    for lang in langs:
+        table += f"| {lang} "
+        for _, num in table_dict[lang].items():
+            table += f"| {num} "
+        table += "|\n"
+
+    return table
+
+
+def insert_tables(file_path: str, tables: list[str], tags: list[str] = ["TASKS TABLE"]):
```
Comment (mrshu) [inline docs/create_tasks_table.py:100]:
```suggestion
def insert_tables(file_path: str, tables: list[str], tags: list[str] = ["TASKS TABLE"]) -> None:
```

File: docs/create_tasks_table.py [hunk 5]
```diff
@@ -56,15 +64,48 @@ def create_tasks_table(tasks: list[mteb.AbsTask]) -> str:
...
+    """Insert tables within <!-- TABLE START --> and <!-- TABLE END --> or similar tags."""
     with open(file_path, "r") as file:
         md = file.read()
```
Comment (mrshu) [inline docs/create_tasks_table.py:104]:
```suggestion
    md = Path(file_path).read_text()
```

File: docs/create_tasks_table.py [hunk 6]
```diff
@@ -56,15 +64,48 @@ def create_tasks_table(tasks: list[mteb.AbsTask]) -> str:
...
-    end = "<!-- TABLE END -->"
-
-    md = md.replace(md[md.index(start) + len(start) : md.index(end)], table)
+    for table, tag in zip(tables, tags):
+        start = f"<!-- {tag} START -->"
+        end = f"<!-- {tag} END -->"
+        md = md.replace(md[md.index(start) + len(start) : md.index(end)], table)
 
     with open(file_path, "w") as file:
         file.write(md)
```
Comment (mrshu) [inline docs/create_tasks_table.py:111]:
```suggestion
    Path(file_path).write_text(md)
```

**Review by mrshu — COMMENTED** [review]
Some small suggestions here and there

File: docs/create_tasks_table.py (hunk 2 above)
Comment (KennethEnevoldsen) [reply docs/create_tasks_table.py:3]: @isaac-chung if this is the case, we can add to the ruff CI that it should just remove unused imports

File: docs/create_tasks_table.py (hunk 1 above)
Comment (KennethEnevoldsen) [reply docs/create_tasks_table.py:70]: I would add 1 for each language in a cross-lingual dataset as well.

Comment (KennethEnevoldsen) [reply docs/create_tasks_table.py:70]: otherwise all bitext becomes zero which seems odd.

File: docs/tasks.md [hunk 7]
```diff
@@ -439,4 +439,256 @@ The following tables gives you an overview of the tasks in MTEB.
 | [YahooAnswersTopicsClassification](https://huggingface.co/datasets/yahoo_answers_topics) | ['eng'] | Classification | s2s | [Web] | {'test': 60000} | {'test': 346.35} |
 | [YelpReviewFullClassification](https://arxiv.org/abs/1509.01626) (Zhang et al., 2015) | ['eng'] | Classification | s2s | [Reviews] | {'test': 50000} |  |
 | [YueOpenriceReviewClassification](https://github.com/Christainx/Dataset_Cantonese_Openrice) (Xiang et al., 2019) | ['yue'] | Classification | s2s | [Reviews] | {'test': 6161} | {'test': 173.0} |
-<!-- TABLE END -->
\ No newline at end of file
+<!-- TASKS TABLE END -->
+
+<br/>
+
+## Task Count per Language
```
Comment (KennethEnevoldsen) [inline docs/tasks.md:446]: I would probably move this above the first table or enclose both tables in a fold down menu (probably the best solution).

File: docs/tasks.md [hunk 8]
```diff
@@ -439,4 +439,256 @@ The following tables gives you an overview of the tasks in MTEB.
...
+| bul | 0 | 4 | 0 | 0 | 1 | 1 | 0 | 1 | 0 | 0 |
+| cat | 0 | 1 | 0 | 0 | 0 | 0 | 0 | 1 | 0 | 0 |
+| ceb | 0 | 1 | 0 | 0 | 0 | 0 | 0 | 1 | 0 | 0 |
+| ces | 0 | 2 | 0 | 0 | 1 | 0 | 0 | 1 | 0 | 0 |
+| chv | 0 | 1 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 |
+| cjk | 0 | 1 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 |
+| ckb | 0 | 1 | 0 | 0 | 0 | 0 | 0 | 1 | 0 | 0 |
+| cmn | 0 | 10 | 4 | 0 | 0 | 3 | 4 | 10 | 8 | 0 |
+| cmo | 0 | 2 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 |
+| code | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 19 | 0 | 0 |
```
Comment (KennethEnevoldsen) [inline docs/tasks.md:504]: I would probably move all code to a separate table

File: docs/tasks.md [hunk 9]
```diff
@@ -439,4 +439,256 @@ The following tables gives you an overview of the tasks in MTEB.
...
+| vie | 1 | 6 | 0 | 0 | 0 | 1 | 0 | 3 | 0 | 0 |
+| war | 0 | 1 | 0 | 0 | 0 | 0 | 0 | 1 | 0 | 0 |
+| wol | 0 | 1 | 0 | 0 | 0 | 0 | 0 | 1 | 0 | 0 |
+| xho | 0 | 3 | 2 | 0 | 0 | 0 | 0 | 1 | 0 | 0 |
+| ydd | 0 | 1 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 |
+| yor | 0 | 3 | 2 | 0 | 0 | 0 | 0 | 1 | 0 | 0 |
+| yue | 0 | 2 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 |
+| zho | 0 | 2 | 0 | 0 | 0 | 1 | 0 | 5 | 0 | 0 |
+| zsm | 0 | 1 | 0 | 0 | 0 | 0 | 0 | 1 | 0 | 0 |
+| zul | 0 | 3 | 0 | 0 | 0 | 0 | 0 | 1 | 0 | 0 |
```
Comment (KennethEnevoldsen) [inline docs/tasks.md:693]: Do we want a total at the end (or at the top)

**Review by KennethEnevoldsen — APPROVED** [review]
Only good only minor suggestions. Only the cross-lingual one seems like a major one.

File: docs/create_tasks_table.py (hunk 2 above)
Comment (isaac-chung) [reply docs/create_tasks_table.py:3]: It's already part of the checks:
```yaml
[tool.ruff.lint]
select = ["F", "I", "E", "D"]
```

Note that that line is not part of the changes, i.e. not highlighted.

File: docs/tasks.md (hunk 9 above)
Comment (isaac-chung) [reply docs/tasks.md:693]: At the end works.

File: docs/create_tasks_table.py (hunk 2 above)
Comment (mrshu) [reply docs/create_tasks_table.py:3]: Sorry about that, my bad. It is indeed used within the other functions that are not highlighted.

File: docs/tasks.md (hunk 7 above)
Comment (isaac-chung) [reply docs/tasks.md:446]: Will do expand for both.

File: docs/create_tasks_table.py (hunk 1 above)
Comment (isaac-chung) [reply docs/create_tasks_table.py:70]: Adding back cross lingual datasets.

File: docs/tasks.md (hunk 9 above)
Comment (isaac-chung) [reply docs/tasks.md]: Added at the end.

**Comment by isaac-chung** [comment]
@KennethEnevoldsen how many points would this be?

File: docs/tasks.md (hunk 8 above)
Comment (isaac-chung) [reply docs/tasks.md:669]: Let's do this in a separate PR?

**Comment by imenelydiaker** [comment]
> @KennethEnevoldsen how many points would this be? and what key 🤔

I added features under the "Bug fixes" key

**Comment by isaac-chung** [comment]
This is really docs / docs that relate to the paper.

**Comment by imenelydiaker** [comment]
So do it under "Paper writing", since it's related

File: docs/tasks.md (hunk 7 above)
Comment (isaac-chung) [reply docs/tasks.md:465]: Done.

**Pull request merged**